### ISSUE TITLE
`zig build` from `nix` cross-compiling doesn't work

### ISSUE THREAD

**Issue by Radvendii** [ziglang/zig#15187]
### Zig Version

0.11.0-dev.2371+a31450375

### Steps to Reproduce and Observed Behavior

Zig can't find dependencies when cross-compiling inside a Nix build.
From what I can tell, Zig has special-casing for nix support where it reads the env variables `NIX_CFLAGS_COMPILE`, and `NIX_LDFLAGS`. 

This happens in [NativePaths.zig](https://github.com/ziglang/zig/blob/master/lib/std/zig/system/NativePaths.zig#L29-L89)

The problem is that this code only runs if we are not cross-compiling ([called here](https://github.com/ziglang/zig/blob/master/src/main.zig#L2513-L2516))

But nix builds set those environment variables even when cross-compiling

Minimal reproduction, including a `meson.build` that does not have the same problem: https://github.com/Radvendii/zig-nix-cross/tree/master

### Expected Behavior

cross-compilation inside a nix package would work.

**Comment by Radvendii** [comment]
Nix has lots of special logic it uses to wrap the c compiler for builds. Usually the build system will use the version nix provides, but zig is its own c compiler as well as build system.

I don't know if there's a way to separate those two functions so that nix can wrap the c compiler part, since reproducing that logic might be fragile (as we're seeing here).

**Comment by Radvendii** [comment]
I might have misunderstood what `cross_target.isNativeOs()` is doing.

**Comment by Radvendii** [comment]
Just found [this comment in build.zig](https://github.com/ziglang/zig/blob/master/build.zig#L722-L725) Not sure if it's related. Also, it's from 3 years ago.

**Comment by Radvendii** [comment]
When I add code to `build.zig` to grab the `NIX_CFLAGS_COMPILE` and `NIX_LDFLAGS` and process them like how they are in zig's own code, it works. Check out [the `workaround` branch](https://github.com/radvendii/zig-nix-cross/tree/workaround) on my minimal example.

This suggests that indeed those flags aren't getting processed when cross-compiling.

The reason this confuses me is that when I add this code into `build.zig`, it prints `true`
```zig
    const cross_target = std.zig.CrossTarget.parse(.{
        .arch_os_abi = "native",
    }) catch unreachable;
    std.log.warn("{any}", .{cross_target.isNativeOs()});
```
so I guess `sysroot != null or (system_libs.count() == 0 and !want_native_include_dirs)`

**Comment by Radvendii** [comment]
Now I'm wondering if the real problem is that it's not using `pkg-config` to find dependencies.

`PKG_CONFIG_PATH` is correctly set to contain the directory of `libconfig.pc`, which should tell it where to find the header files.

The trouble is I'm not sure what the expected / correct behaviour is.

**Comment by Radvendii** [comment]
I tried all this again recently from scratch and have managed to get it working. I don't remember exactly what the issue was to begin with, but something I did this time that might have been related was creating a `pkg-config` executable in the `PATH` that points at the target-specific `pkg-config`. I'm not sure how things are **supposed** to work in nix, but there's no copy of `pkg-config` that gets added to `PATH` **with that name** when cross-compiling.

```nix
nativeBuildInputs = [ 
  pkg-config
  (buildPackages.runCommand "rename-pkg-config" {} ''
    mkdir -p $out/bin
    ln -s ${buildPackages.pkg-config}/bin/x86_64-w64-mingw32-pkg-config $out/bin/pkg-config
  '')
];
```

Note that `buildPackages.` might not be necessary.
